### ISSUE TITLE
Add Pagination Parameters to DAG List API

### ISSUE THREAD

**Issue by yohamta** [daguflow/dagu#653]
**Overview**
The DAG list API can be slow when there are a large number of DAG files because the API reads all YAML files in the directory.

To address this issue, we would like to add `page` and `limit `parameters to the `/api/dags/v1` API to limit the response.

**Requirements**
- Update the [swagger file](https://github.com/daguflow/dagu/blob/main/swagger.yaml) to include new parameters.
- Update the [list API implementation](https://github.com/daguflow/dagu/blob/main/internal/frontend/dag/handler.go#L171).
- Update the [frontend code](https://github.com/daguflow/dagu/blob/main/ui/src/pages/dags/index.tsx) to include the pagination UI.

**Comment by halalala222** [comment]
![image](https://github.com/user-attachments/assets/c54e3750-0a0d-4dd1-b544-18dda21e556a)
add a pagination
like this ?
![image](https://github.com/user-attachments/assets/33c864e4-2d9a-4a43-92d5-53cf152400a3)
I want to know the value of the pageSize (limit).

**Comment by yohamta** [comment]
Thank you very much! I believe setting the default value to 50 should be sufficient.

I just realized that this enhancement also requires adding query parameters for names and tags. Currently, DAG filtering is handled by JavaScript in the browser, so we need to shift this functionality to the server side. This is necessary because, with server-side pagination, the server cannot determine how many DAGs match a particular page.

**Comment by halalala222** [comment]
Yes,I also thought of names and tags filtering and was planning to bring it up, but I went to eat and didn't mention it.

**Comment by halalala222** [comment]
I will try to work on this enhancement

**Comment by halalala222** [comment]
@yohamta while create dag will not contain the tag value
![image](https://github.com/user-attachments/assets/34524dbc-c69f-426f-9f57-c8ecbf602724)
I also haven't found how to modify the tag.

**Comment by halalala222** [comment]
@yohamta 
I also find a big
![image](https://github.com/user-attachments/assets/73c2934d-ab24-4884-92c6-001820b54397)
![image](https://github.com/user-attachments/assets/27dd34f9-e60b-4e0a-9538-faddcf2c2a72)
I will also fix this minor bug.

**Comment by yohamta** [comment]
@halalala222 Thank you very much!

> @yohamta while create dag will not contain the tag value
I als can't found how to modify the tag.

You can set the tag value in the yaml like:
```yaml
tags: monthly
steps
  - name: step1
    command: echo hello
```

**Comment by halalala222** [comment]
> @halalala222 Thank you very much!
> 
> > @yohamta while create dag will not contain the tag value
> > I als can't found how to modify the tag.
> 
> You can set the tag value in the yaml like:
> 
> ```yaml
> tags: monthly
> 
> steps:
>   - name: step1
>     command: echo hello
> ```

thanks you !

**Comment by codinggeeks06** [comment]
hey @halalala222 and @yohamta,
i really want to work on this issue. if you folks have no problem with it. can i work on it?

**Comment by codinggeeks06** [comment]
hey,
no worries. i only wanted to approach it if it hasn't been picked up, looking forward to this feature. yay!
